airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/job_error_reporter/SentryExceptionHelper.java

```diff
@@ -166,4 +169,43 @@ private static Optional<List<SentryException>> buildJavaSentryExceptions(final S
     return Optional.of(sentryExceptions);
   }
 
+  private static Optional<List<SentryException>> buildNormalizationDbtSentryExceptions(final String stacktrace) {
+    final List<SentryException> sentryExceptions = new ArrayList<>();
+
+    // focus on Database Errors (these are the common errors seen in Airbyte)
+    final String databaseErrorIdentifier = "Database Error in model";
+    // for other dbt error types, we'll default to non-parsed stacktrace grouping
+    if (!stacktrace.contains(databaseErrorIdentifier)) {
+      return Optional.empty();
+    }
+
+    final String[] separatedStackTrace = stacktrace.split("\n");
+
+    // first let's get our useful error line for grouping
+    String usefulError = "";
+    boolean nextLine = false;
+    for (String errorLine : separatedStackTrace) {
+      // previous line was "Database Error..." so this is our useful message line
+      if (nextLine) {
+        usefulError = errorLine;
+        break;
+      }
+      if (errorLine.contains("Database Error in model")) {
+        nextLine = true;
+      }
+    }
+
+    if (!"".equals(usefulError)) {
+      final SentryException usefulException = new SentryException();
+      usefulException.setValue(usefulError);
+      usefulException.setType("DbtDatabaseError");
```

Comment:
Would it be useful to add the dbt error code to the type? It may help separate the issues more since they're all the same `DbtDatabaseError` type 🤔 Sentry also tries to be smart and replace certain things like numbers from the exception value which may or may not remove the error codes

Reply:
There _are_ error codes within the log jsons themselves, although I can't find any lookup reference in dbt's docs for them. I think [my comment here](https://github.com/airbytehq/airbyte/pull/15695#issuecomment-1219839942) is relevant (in terms of differentiating issues better), but open to a future change adding in the log code for differentiation if it becomes an issue.

 I think it may be okay since the message is also shown on each issue (and they're grouped by that not the Exception type/title) e.g. 
<img width="829" alt="Screenshot 2022-08-18 at 20 06 02" src="https://user-images.githubusercontent.com/27702686/185474584-befff2e8-d753-44a9-a171-ce2f892cddea.png">

Reply:
Makes sense! I couldn't find what the error codes mean either lol